benches/rsexp_benchmark.rs

```diff
@@ -0,0 +1,67 @@
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use rand::seq::SliceRandom;
+use rand::thread_rng;
+
+fn make_n_random_characters(n: i64, alphabet: &Vec<char>) -> String {
+    let mut rng = thread_rng();
+    (0..n).map(|_| alphabet.choose(&mut rng).unwrap()).collect()
+}
+
+fn make_benchmark_string(
+    num_repetitions: &[i64],
+    str_len: i64,
+    quoted: bool,
+    alphabet: &Vec<char>,
+) -> String {
+    if let Some(len) = num_repetitions.get(0) {
+        format!(
+            "({})",
+            std::iter::repeat(make_benchmark_string(
+                &num_repetitions[1..],
+                str_len,
+                quoted,
+                alphabet
+            ))
+            .take(*len as usize)
+            .collect::<String>()
+        )
+    } else {
+        let chars = make_n_random_characters(str_len, alphabet);
+        if quoted {
+            format!("\"{}\"", chars)
+        } else {
+            chars
+        }
+    }
+}
+
+fn parse_sexp(contents: &[u8]) {
+    let _sexp = rsexp::from_slice(&contents).unwrap();
+}
+
+pub fn criterion_benchmark(c: &mut Criterion) {
+    let alphabet: Vec<char> = (b'a'..=b'z').map(char::from).collect();
+
+    for quoted in [true, false] {
+        for (str_len, repetitions, depth) in
+            [(4, 100, 1), (4, 100, 2), (10, 100, 2), (1000, 100, 1)]
+        {
+            let bench_name = format!(
+                "{repetitions}_repetitions_{depth}_depth_{str_len}_strlen_{quoted}",
+                repetitions = repetitions,
+                depth = depth,
+                str_len = str_len,
+                quoted = (if quoted { "quoted" } else { "unquoted" })
```

Comment:
No need to do this in this PR but a nice separate benchmark to add would be strings that actually have to be quoted because they use some escape sequences.
These could easily be generated by using a wider alphabet (e.g. any byte) and using the `Sexp::Atom(...)::to_string`. Maybe the whole sexp could be generated randomly and `to_string` applied only once at the end. It's actually what is done in the tests via quickcheck with a probability 1/2 for each string to be made of alphabetical characters vs 1/2 to be random bytes, see details [here](https://github.com/LaurentMazare/rsexp/blob/c334749662c144218b1feef79dfa78898ef488e5/tests/sexp_tests.rs#L9-L24).

Reply:
I think it's a great idea to construct the sexp natively + include escape sequences. Added a TODO for the escape sequences here.